src/components/pages/home.jsx

```diff
@@ -1,45 +1,59 @@
 import {Link} from "react-router"
-import React from "react"
+import React, {PropTypes} from "react"
 
 import BreadCrumb from "../breadcrumb"
+import ExternalLink from "../external-link"
 
 
 const HomePage = React.createClass({
+  propTypes: {
+    countryPackageName: PropTypes.string.isRequired,
```

Comment:
Here are the steps:
- On the server-side, the express server downloads the `/api/1/parameters` and `/api/1/variables` endpoints in parallel via `Promise.all`: [see here](https://github.com/openfisca/legislation-explorer/blob/8a8e9660a5d28552309e0d1c15887da354eff82a/src/server/index.js#L65-L84), then [writes the result to `window.__INITIAL_STATE__`](https://github.com/openfisca/legislation-explorer/blob/8a8e9660a5d28552309e0d1c15887da354eff82a/src/server/html-document.jsx#L26-L28).
- On the client side, the initial state is read from `window.__INITIAL_STATE__`: [see here](https://github.com/openfisca/legislation-explorer/blob/8a8e9660a5d28552309e0d1c15887da354eff82a/src/client.jsx#L16) then passed to the router which calls the right page depending on the URL. The router is in charge to pass these "properties" to the component. See routes [here](https://github.com/openfisca/legislation-explorer/blob/8a8e9660a5d28552309e0d1c15887da354eff82a/src/routes.jsx)

I copied this workflow from documentations of React-Router, that's a quite known way to do with single-page applications in React, and YES it's over-complicated (that's one of the reasons I left those technologies...).